### ISSUE TITLE
auto testing, or semi-automatic testing of Materials and derivatives

### ISSUE THREAD

**Issue by WilkAndy** [WilkAndy/moose#26]
In @cpgr latest PR he demonstrated how a coder could test that density, viscosity, etc produced the expected output.  We can easily do the same for the derivatives of them, and should construct some test files showing this.  This is "semi-automatic" mode, which i'm sure the Dictator approves of.

It'd be even nicer if we could rig up some automatic testing (like -snes_type=test), but that is not high-priority unless someone can see an easy way of doing it.

**Comment by cpgr** [comment]
Yep, can easily test the derivatives in the same test. Just didn't put them in at present. I'll look after this
